src/main/java/io/github/mountainrange/mule/enums/MapType.java

```diff
@@ -4,36 +4,86 @@
  * List of map types (for the map generator).
  */
 public enum MapType {
-	CLASSIC, RANDOM, EXPERIMENTAL, EMPTY;
+	CLASSIC, RANDOM_MED, EXPERIMENTAL, EMPTY;
 
-	public final TerrainType[][] map;
+	private TerrainType[][] map;
 
-	MapType() {
-		// Abbreviations
-		final TerrainType T99 = null;
-		final TerrainType T00 = TerrainType.PLAIN;
-		final TerrainType T01 = TerrainType.RIVER;
-		final TerrainType T02 = TerrainType.MOUNTAIN1;
-		final TerrainType T03 = TerrainType.MOUNTAIN2;
-		final TerrainType T04 = TerrainType.MOUNTAIN3;
-		final TerrainType T05 = TerrainType.TOWN;
-
-		if (this.name().equalsIgnoreCase("empty")) {
-			this.map = new TerrainType[][]{
+	private static final TerrainType[][] EMPTY_MAP;
+	private static final TerrainType[][] DEFAULT_MAP;
+
+	// Abbreviations
+	private static final TerrainType T99 = null;
+	private static final TerrainType T00 = TerrainType.PLAIN;
+	private static final TerrainType T01 = TerrainType.RIVER;
+	private static final TerrainType T02 = TerrainType.MOUNTAIN1;
+	private static final TerrainType T03 = TerrainType.MOUNTAIN2;
+	private static final TerrainType T04 = TerrainType.MOUNTAIN3;
+	private static final TerrainType T05 = TerrainType.TOWN;
+
+	static {
+		EMPTY_MAP = new TerrainType[][]{
 				{T99, T99, T99, T99, T99, T99, T99, T99, T99},
 				{T99, T99, T99, T99, T99, T99, T99, T99, T99},
 				{T99, T99, T99, T99, T99, T99, T99, T99, T99},
 				{T99, T99, T99, T99, T99, T99, T99, T99, T99},
 				{T99, T99, T99, T99, T99, T99, T99, T99, T99}
-			};
-		} else {
-			this.map = new TerrainType[][]{
+		};
+
+		DEFAULT_MAP = new TerrainType[][]{
 				{T00, T00, T02, T00, T01, T00, T04, T00, T00},
 				{T00, T02, T00, T00, T01, T00, T00, T00, T04},
 				{T04, T00, T00, T00, T05, T00, T00, T00, T02},
 				{T00, T03, T00, T00, T01, T00, T03, T00, T00},
 				{T00, T00, T03, T00, T01, T00, T00, T00, T03}
-			};
+		};
+	}
+
+	MapType() {
+	}
+
+
+	private static TerrainType[][] generateRandomMap(int cols, int rows) {
+		int riverPosition = (int) (Math.random() * cols);
+		TerrainType[][] tileGrid = new TerrainType[rows][cols];
+		for (int y = 0; y < rows; y++) {
+			for (int x = 0; x < cols; x++) {
+				double r = Math.random();
+				if (x == (cols/2) && y == (rows/2)) {
+					tileGrid[y][x] = T05;
+				} else if (x == riverPosition) {
```

Comment:
I changed the way that the river generation worked to be a bit smarter, rather than scattering river tiles everywhere
I left the old code in for reference though.